.github/workflows/main.yml

```diff
@@ -0,0 +1,64 @@
+name: Test Suite
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        orm: [active_record, mongoid]
+        rails: ['4.2', '5.0', '5.1', '5.2', '6.0']
+        ruby: ['2.5', '2.6', '2.7'] # TODO fix repo to work with ruby-head
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+    - uses: actions/cache@v2
+      with:
+        path: |
+          **/bundle
+        key: bundle-use-ruby-${{ matrix.ruby }}-rails-${{ matrix.rails }}-orm-${{ matrix.orm }}-gemfile-${{ hashFiles(format('**/gemfiles/rails_{0}.gemfile', matrix.rails)) }}-gemspec-${{ hashFiles('**/devise-security.gemspec') }}
+    - name: bundle install
+      env:
+        BUNDLE_GEMFILE: gemfiles/rails_${{ matrix.rails }}.gemfile
+      run: |
+        bundle config path vendor/bundle
+        bundle install --jobs 4
+      if: ${{ matrix.rails != '4.2' || (matrix.rails == '4.2' && matrix.ruby == '2.5') }}
+    - name: Start MongoDB
+      uses: superchargejs/mongodb-github-action@1.0.0
+      if: ${{ matrix.orm == 'mongoid' && (matrix.rails != '4.2' || (matrix.rails == '4.2' && matrix.ruby == '2.5')) }}
+    - name: Tests for ORM ${{ matrix.orm }}, Rails ${{ matrix.rails }}, and Ruby ${{ matrix.ruby }}
+      env:
+        BUNDLE_GEMFILE: gemfiles/rails_${{ matrix.rails }}.gemfile
+        CI: true
+        DEVISE_ORM: ${{ matrix.orm }}
+      run: |
+        bundle exec rake
+      if: ${{ matrix.rails != '4.2' || (matrix.rails == '4.2' && matrix.ruby == '2.5') }}
+    - name: Coveralls Parallel
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        flag-name: run-orm-${{ matrix.orm }}-rails-${{ matrix.rails }}-ruby-${{ matrix.ruby }}
+        parallel: true
+        path-to-lcov: ./coverage/lcov/devise-security.lcov
+      if: ${{ matrix.rails != '4.2' || (matrix.rails == '4.2' && matrix.ruby == '2.5') }}
+
+  finish:
+    needs: run-tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Coveralls Finished
+      uses: coverallsapp/github-action@master
```

Comment:
Based on their [docs](https://github.com/marketplace/actions/coveralls-github-action), it looks like this Action should add a comment with test coverage information so it seems like we shouldn't miss out on anything from Coveralls 👍  (not sure if we did anything with checking for a certain percentage of coverage).

Reply:
https://github.com/devise-security/devise-security/pull/252#issuecomment-714220480 that's what happened here? Unless you're referring to something else?